### DISCUSSION TITLE
Is it safe to delete/update documents in jv_snapshot manually?

### DISCUSSION THREAD

**Discussion by abhinavDCT** [javers/javers#1334] [Q&A] [answered]
We have been tracking changes for some of the collections with Javers (and Spring Boot and MongoDB).

1. We wanted to delete some of the documents in jv_snapshot as we unintentionally audited them (as we did not add @DiffIgnore earlier).
2. We also wanted to migrate author field in commitMetadata, which is currently storing email to UUID.

Is it safe to make these changes directly in the database using script or could it create some isssues?

**Comment by bartoszwalacik** [comment, marked as answer]
1. It should be safe, but please remember about two rules of thumb
> delete from tail to head, so starting from the oldest snapshots
> snapshots stores references between objects, don't leave dangling references, by deleting subsets of snapshot graphs
2. you can change the author field freely, it's just a label
